llvm/lib/Transforms/IPO/RepoPruning.cpp

```diff
@@ -123,14 +123,61 @@ repodefinition::DigestType toDigestType(pstore::index::digest D) {
   return Digest;
 }
 
+/// Two kinds of pruning exist in the repository level and the module level. The
+/// repository level pruning: if the global object is compiled before and exists
+/// in the repository, it will be pruned. The module level pruning: in the same
+/// compilation unit (module), some global objects have the same digests. The
+/// RepoPruning pass could prune them as well.
+///
+/// \param Repository The database.
+/// \param FragmentIndex A database fragment index.
+/// \Key the GO's digest.
+/// \param GO A global object.
+/// \returns true if the GO is in the existing repository.
+static bool hasExistingGOInRepository(
+    const pstore::database &Repository,
+    const std::shared_ptr<const pstore::index::fragment_index> &FragmentIndex,
+    const pstore::index::digest &Key, const GlobalObject &GO) {
+  if (FragmentIndex &&
+      FragmentIndex->find(Repository, Key) != FragmentIndex->end(Repository)) {
+    LLVM_DEBUG(dbgs() << "Existing GO (in repository) name: " << GO.getName()
+                      << '\n');
+    return true;
+  }
+  return false;
+}
+
+/// \param ModuleFragments The collection of fragments that will appear in this
+///        compilation but not include the pruned entries.
+/// \Key the GO's digest.
```

Comment:
As above. clang will warn that \Key is not a valid Doxygen comment (as will, of course, Doxygen itself).

Reply:
Fixed, thanks.